src/Tasks/Common/build/Microsoft.NET.DefaultPackageConflictOverrides.targets

```diff
@@ -0,0 +1,233 @@
+<!--
+***********************************************************************************************
+Microsoft.NET.DefaultPackageConflictOverrides.targets
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (c) .NET Foundation. All rights reserved. 
+***********************************************************************************************
+-->
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <PropertyGroup>
+    <MSBuildAllProjects>$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
+  </PropertyGroup>
+
+  <ItemGroup Condition="'$(DisableDefaultPackageConflictOverrides)' != 'true'">
+    <PackageConflictOverrides Include="Microsoft.NETCore.App">
+      <OverridenPackages>
+        Microsoft.CSharp|4.4.0;
```

Comment:
@weshaggard - is there a list I can compare these packages to for both NETCore.App and NETStandard? The tricky thing is which OOB packages need versions `> 4.3.0`.

Reply:
Do what just the latest versions? It would seem to make this a useful cache you would need all versions. 

As for the list of them I think the best you can do is to provide the latest version of everything that lives in Microsoft.NETCore.App and lookup their latest version on nuget.org.

Reply:
We are using [a `<=` check](https://github.com/dotnet/sdk/pull/1805/files#diff-210154c344df4e4338c157d85a780a87R72).  So if `Microsoft.CSharp 4.1.0` package is encountered, it still loses to `Microsoft.NETCore.App`.

Reply:
We need to be a little careful about that type of logic in servicing type of scenarios. During servicing we will have a 4.x.x that we might actually need to override what is in Microsoft.NETCore.App.

Reply:
Concretely, you mean we might ship a `4.3.1` Microsoft.CSharp that should override the `4.4.0` version inside Microsoft.NETCore.App `2.0.0`?

Reply:
I guess thinking about it a little more I think it is pretty unlikely to ship something that overrides the inbox Microsoft.NETCore.App from anything other then the release/2.0.0 branch (i.e. 4.4.x). So we need to at least ensure that these versions are somehow tied to a version of Microsoft.NETCore.App as well. As in we may very well want the 4.4.1 (if and when it ships) version of Microsoft.Csharp to win a conflict over the 2.0.0 Microsoft.NETCore.App but not for the 2.0.x and greater version that contains it. 

I'm not sure how this affects, if at all, Microsoft.NETCore.App < 2.0.0, but we need to be sure that this doesn't start to throw out packages there.

Reply:
My thinking is that these versions in the SDK won't change going forward.  They are the "baseline".

We can do the following going forward in future versions of Microsoft.NETCore.App and/or NETStandard.Library:

1. Provide higher version numbers to the existing packages, and add new packages.  see [MergePackageOverrides](https://github.com/dotnet/sdk/pull/1805/files#diff-210154c344df4e4338c157d85a780a87R44)
1. Set `DisableDefaultPackageConflictOverrides=true` and provide a whole new list of these packages and versions.  Or set that boolean to turn this functionality off all together.
1. Choose to not update these packages/versions in Microsoft.NETCore.App or NETStandard.Library.  This won't cause any correctness issues, since this new check is only a performance optimization.  If a higher Microsoft.CSharp package (say `4.4.1`) conflicts with a future version of Microsoft.NETCore.App, then the existing AssemblyVersion and FileVersion checks will kick in and do the right thing.  They will just happen to be a bit slower than these NuGet version checks.

I'm choosing to go with the last option for now, until we have data that proves we need to add those future versions to Microsoft.NETCore.App and/or NETStandard.Library.  `>90%` of the conflicts are caused by packages that we aren't shipping new versions for (System.Runtime/Console/Collections/IO/etc).  So even if we don't keep the OOB packages up-to-date, we still get a big perf win here.

Reply:
That makes sense to me. 

I'm wondering if there can/should be a test to validate that all of these overrides behave the same as the fallback assemblyversion/fileversion checks. I'm imagining a test that generates a set of package references from the overrides and then we build with and it without DisableDefaultPackageConflictOverrides and check that we get the same `@(Reference)` and `@(ReferenceCopyLocalPaths)` both ways. You can get the overrides and the reference lists with GetValuesCommand.

Reply:
> I'm not sure how this affects, if at all, Microsoft.NETCore.App < 2.0.

I would think that none of the assets out of these packages would ever conflict with Microsoft.NETCore.App or NETStandard.Library < 2.0.0 and so there would be no impact there. Right?

Reply:
Since < 2.0 were packages references nuget should unify the packages and so there shouldn't be a conflict but I not sure if there is any logic to always compare to what is part of Microsoft.NETCore.App so thought I would ask the question just to be sure others think about it as well.